Lib/threading.py

```diff
@@ -839,6 +839,16 @@ class is implemented.
         # For debugging and _after_fork()
         _dangling.add(self)
 
+    def __del__(self):
+        if not self._initialized:
+            return
+        lock = self._tstate_lock
+        if lock is not None and not self.daemon:
+            if lock.acquire(False):
```

Comment:
If the lock is not acquired, then the lock still remains in the global list and leaks.

Reply:
You are right. Running threads will still in interpreter.

```
import sys
import random
import threading
import time

def f():
    seconds = random.random() * 0.01
    time.sleep(seconds)

for _ in range(10):
    threading.Thread(target=f).start()
    print(sys.gettotalrefcount())
```

Reply:
Wait. this test case can not measure the refleaks. It will raise the refleaks too when I apply #15228.